### DISCUSSION TITLE
Downloading PDF in Firefox / Safari failes, Chrome no issues

### DISCUSSION THREAD

**Discussion by timo002** [dompdf/dompdf#2495] [Q&A]
I'm using dompdf 1.0.2 icm with Symfony 4.4. When I create a PDF I have no issues when I download the files with Google Chrome browser. But when using Safari or Firefox, it sometimes works, but in most cases the downloaded file is broken. It will send a file, but not all the data is received.

The response headers in a succesfull or not succesfull call both state the same content-length. But when the download fails, the content is truncated at certain point. 

I don't understand why Chrome is working fine, while Firefox and Safari often download broken PDF.

The code in my controller to download the files is shown below.
```
        $pdfOptions = new Options();
        $pdfOptions->set('defaultFont', 'DejaVu Sans');
        $pdfOptions->setChroot($appKernel->getProjectDir());

        $dompdf = new Dompdf($pdfOptions);

        $html = $this->renderView('pdf/invoice.html.twig', [
            'invoice' => $invoice,
            'projectDir' => $appKernel->getProjectDir(),
        ]);

        $dompdf->loadHtml($html);

        $dompdf->setPaper('A4', 'portrait');

        $dompdf->render();

        $number = ($invoice->getInvoiceNumber() != '') ? $invoice->getInvoiceNumber() : $invoice->getId();

        $dompdf->stream($number . ' ' . date('Y-m-d') . ' ' . $helper->slugify($invoice->getInvoiceAddress()->getCompany(), ' ') . '.pdf', [
            'Attachment' => true
        ]);

        return new Response('PDF file should be downloaded!');
```

**Comment by bsweeney** [comment]
Could be an issue interacting with the web server. Instead of returning a response from your method can you try calling `exit()` after `$dompdf->stream(...)`?

**Comment by fd6130** [comment]
Isn't that you have to return `$dompf->stream(...)` instead of `new Response()` ? Because that is a StreamResponse, you can just return that.
